dacbench/benchmarks/layerwise_sgd_benchmark.py

```diff
@@ -0,0 +1,222 @@
+        "layer_specification": [
+            (
+                sgd_utils.LayerType.CONV2D,
+                {"in_channels": 1, "out_channels": 32, "kernel_size": 3},
+            ),
+            (sgd_utils.LayerType.RELU, {}),
+            (
+                sgd_utils.LayerType.CONV2D,
+                {"in_channels": 32, "out_channels": 64, "kernel_size": 3},
+            ),
+            (sgd_utils.LayerType.RELU, {}),
+            (sgd_utils.LayerType.POOLING, {"kernel_size": 2}),
+            (sgd_utils.LayerType.DROPOUT, {"p": 0.25}),
+            (sgd_utils.LayerType.FLATTEN, {"start_dim": 1}),
+            (
+                sgd_utils.LayerType.LINEAR,
+                {"in_features": 9216, "out_features": 128},
+            ),
+            (sgd_utils.LayerType.RELU, {}),
+            (sgd_utils.LayerType.DROPOUT, {"p": 0.25}),
+            (sgd_utils.LayerType.LINEAR, {"in_features": 128, "out_features": 10}),
+            (sgd_utils.LayerType.LOGSOFTMAX, {"dim": 1}),
+        ],
+        "torch_hub_model": (False, False, False),
+        "optimizer_params": {
+            # "weight_decay": 10.978902603194243,
+            # "eps": 1.2346464628039852e-10,
+            # "betas": (0.9994264825468422, 0.9866804882743139),
+            "momentum": 0.9,
+            "lr": 0.002,
+        },
+        "cutoff": 1e2,
+        "loss_function": __default_loss_function,
+        "loss_function_kwargs": {},
+        "training_batch_size": 64,
+        "fraction_of_dataset": 0.6,
+        "train_validation_ratio": 0.8,  # If set to None, random value is used
+        "dataset_name": "MNIST",  # If set to None, random data set is chosen;
+        # else specific set can be set: e.g. "MNIST"
+        # "reward_function":,    # Can be set, to replace the default function
+        # "state_method":,       # Can be set, to replace the default function
+        "seed": 0,
+        "crash_penalty": -10000.0,
+        "initial_learning_rate": 0.002,
+        "multi_agent": False,
+        "instance_set_path": "../instance_sets/sgd/sgd_train_100instances.csv",
+        "benchmark_info": INFO,
+        "dataset": "MNIST",  # MNIST, CIFAR10, FashionMNIST
+        "epoch_mode": False,
+        "instance_mode": "random_seed",
+    }
+)
+
+
+class LayerwiseSGDBenchmark(AbstractBenchmark):
+    """Benchmark with default configuration & relevant functions for SGD."""
+
+    def __init__(self, config_path=None, config=None):
+        """Initialize layerwise SGD Benchmark.
+
+        Parameters
+        -------
+        config_path : str
+            Path to config file (optional)
+        """
+        super().__init__(config_path, config)
+        if not self.config:
+            self.config = objdict(LAYERWISE_SGD_DEFAULTS.copy())
+
+        for key in LAYERWISE_SGD_DEFAULTS:
+            if key not in self.config:
+                self.config[key] = LAYERWISE_SGD_DEFAULTS[key]
+            elif key == "instance_mode" and self.config[key] == "":
```

Comment:
these if clauses can be combined with an or. Don't know how readable that is though